cypress/e2e/reviewRequests.spec.ts

```diff
@@ -108,127 +157,510 @@ describe("Review Requests", () => {
+      const MERGE_INTERCEPTOR = "getReviewRequest"
+      cy.intercept("POST", "**/merge").as(MERGE_INTERCEPTOR)
+      closeReviewRequests()
+      cy.setEmailSessionDefaults("Email admin")
+      editUnlinkedPage(
+        "faq.md",
+        "some reviewer merge review request content",
+        E2E_EMAIL_TEST_SITE.repo
+      )
+      createReviewRequest(
+        MOCK_REVIEW_TITLE,
+        [E2E_EMAIL_ADMIN_2.email],
+        MOCK_REVIEW_DESCRIPTION
+      ).then((id) => {
+        cy.actAsEmailUser(E2E_EMAIL_ADMIN_2.email, "Email admin")
+        approveReviewRequest(id)
+      })
+      cy.actAsEmailUser(E2E_EMAIL_ADMIN.email, "Email admin")
+      visitE2eEmailTestRepo()
+
       // Act
+      getViewReviewRequestButton().click()
+      cy.wait(`@${RR_INTERCEPTOR}`)
+      getPublishButton().click()
+      cy.wait(`@${MERGE_INTERCEPTOR}`)
+
       // Assert
+      cy.contains("Your changes have been published!").should("exist")
     })
   })
-  describe.skip("changing review request states", () => {
+  describe("changing review request states", () => {
+    before(() => {
+      addAdmin(E2E_EMAIL_ADMIN_2.email)
+      closeReviewRequests()
+    })
+    beforeEach(() => {
+      // NOTE: We want to start this test suite on a clean slate
+      // and prevent other tests from interfering
+      cy.setupDefaultInterceptors()
+      cy.setEmailSessionDefaults("Email admin")
+      closeReviewRequests()
+      visitE2eEmailTestRepo()
+    })
     it("should allow the reviewer to unapprove the request", () => {
       // Arrange
+      const RR_INTERCEPTOR = "getReviewRequest"
+      cy.intercept("GET", "**/summary").as(RR_INTERCEPTOR)
+      cy.setEmailSessionDefaults("Email admin")
+      cy.actAsEmailUser(E2E_EMAIL_ADMIN_2.email, "Email admin")
+      editUnlinkedPage(
+        "faq.md",
+        "some reviewer unapprove review request content",
+        E2E_EMAIL_TEST_SITE.repo
+      )
+      createReviewRequest(
+        MOCK_REVIEW_TITLE,
+        [E2E_EMAIL_ADMIN.email],
+        MOCK_REVIEW_DESCRIPTION
+      ).then((id) => {
+        cy.actAsEmailUser(E2E_EMAIL_ADMIN.email, "Email admin")
+        approveReviewRequest(id)
+      })
+      visitE2eEmailTestRepo()
+
       // Act
+      getViewReviewRequestButton().click()
+      cy.wait(`@${RR_INTERCEPTOR}`)
+      openReviewRequestStateDropdown("Approved")
+      cy.contains("In review").should("exist")
+      cy.contains("In review").click()
+      // Wait for dropdown menu to disappear
+      cy.wait(1000)
+
       // Assert
+      cy.contains("button", "Approved").should("not.be.visible")
+      cy.contains("button", "In review").should("be.visible")
     })
     it("should allow the requestor to close the review request", () => {
       // Arrange
+      const RR_INTERCEPTOR = "getReviewRequest"
+      cy.intercept("GET", "**/summary").as(RR_INTERCEPTOR)
+      const RR_CANCEL_INTERCEPTOR = "getReviewRequest"
+      const regex = /\/review\/\d+\//
```

Comment:
sorry, could i get a brief explanation on this?

Reply:
Have added a comment in 74c94e5f!